pkgs/desktops/gnome/misc/gnome-flashback/default.nix

```
@@ -155,15 +155,38 @@ let
             '';
           };
 
-          executable = writeShellScriptBin "gnome-flashback-${wmName}" ''
-            if [ -z $XDG_CURRENT_DESKTOP ]; then
-              export XDG_CURRENT_DESKTOP="GNOME-Flashback:GNOME"
-            fi
-
-            export XDG_DATA_DIRS=${wmApplication}/share:${gnomeSession}/share:${gnome-flashback}/share:${gnome-panel}/share:$XDG_DATA_DIRS
+          # gnome-panel will only look for applets in a single directory so symlink them into here.
+          panelModulesEnv = buildEnv {
+            name = "panel-modules-env";
+            # We always want to find the built-in panel applets.
+            paths = [ gnome-panel ] ++ panelModulePackages;
+            pathsToLink = [ "/lib/gnome-panel/modules" ];
+          };
 
-            exec ${gnome-session}/bin/gnome-session --session=gnome-flashback-${wmName} "$@"
-          '';
+          executable = stdenv.mkDerivation {
+            name = "gnome-flashback-${wmName}";
+            nativeBuildInputs = [ wrapGAppsHook ];
+            buildInputs = [ glib gnome-panel gnome.gnome-flashback ] ++ panelModulePackages;
+
+            # We want to use the wrapGAppsHook mechanism to wrap gnome-session
+            # with the environment that gnome-flashback and gnome-panel need to
+            # run, including the configured applet packages. This is only possible
+            # in the fixup phase, so turn everything else off.
+            dontUnpack = true;
+            dontConfigure = true;
+            dontBuild = true;
+            dontInstall = true;
+            dontWrapGApps = true; # We want to do the wrapping ourselves.
+
+            preFixup = ''
+              makeWrapper ${gnome-session}/bin/gnome-session $out \
+                --add-flags "--session=gnome-flashback-${wmName}" \
+                --set XDG_CURRENT_DESKTOP "GNOME-Flashback:GNOME" \
```

Not sure why the original script only defaulted to this value but we might want to keep that, Chesterton’s fence and all.

I would try

```suggestion
                --set XDG_CURRENT_DESKTOP "''${XDG_CURRENT_DESKTOP:-GNOME-Flashback:GNOME}" \
```

and, if that does not work due to escaping, `--set-default`. (Though that one behaves differently when the variable is set but empty.